--- VertexShaderGeneratorGLES_orig.cpp	2019-10-11 15:24:43.114947000 +0200
+++ VertexShaderGeneratorGLES.cpp	2019-10-14 18:15:59.232015463 +0200
@@ -598,6 +598,13 @@
 
 		WRITE(p, "  vec4 viewPos = u_view * vec4(worldpos, 1.0);\n");
 
+		const float fov_factor = 2.0;
+		const float z_offset = 550.0;
+
+		WRITE(p, "  viewPos.x /= %f;\n", fov_factor);
+		WRITE(p, "  viewPos.y /= %f;\n", fov_factor);
+		WRITE(p, "  viewPos.z += %f*viewPos.w;\n", z_offset);
+
 		// Final view and projection transforms.
 		if (gstate_c.Supports(GPU_ROUND_DEPTH_TO_16BIT)) {
 			WRITE(p, "  vec4 outPos = depthRoundZVP(u_proj * viewPos);\n");
